src/pages/Dashboard/Modals/SupplyWithdraw/SupplyWithdrawForm.tsx

```diff
@@ -0,0 +1,201 @@
+  formatCentsToReadableValue,
+  format,
+  currencyFormatter,
+} from 'utilities/common';
+import { useStyles } from '../styles';
+
+interface ISupplyWithdrawFormProps extends FormikProps<FormValues> {
+  asset: Asset;
+  tokenInfo: IInfoRowProps['info'];
+  userTotalBorrowBalance: BigNumber;
+  userTotalBorrowLimit: BigNumber;
+  dailyEarnings: BigNumber;
+  onSubmit: IAmountFormProps['onSubmit'];
+  enabledButtonKey: string;
+  disabledButtonKey: string;
+  maxInput: BigNumber;
+  newBalance: (amount: BigNumber) => BigNumber;
+}
+
+export const SupplyWithdrawForm: React.FC<ISupplyWithdrawFormProps> = ({
+  values,
+  setFieldValue,
+  resetForm,
+  asset,
+  tokenInfo,
+  userTotalBorrowBalance,
+  userTotalBorrowLimit,
+  dailyEarnings,
+  maxInput,
+  enabledButtonKey,
+  disabledButtonKey,
+  newBalance,
+}) => {
+  const { id: assetId } = asset;
+  const { amount } = values;
+  const validAmount = amount && !amount.isZero() && !amount.isNaN();
+  const styles = useStyles();
+  const { t, Trans } = useTranslation();
+  const [newBorrowLimit, setNewBorrowLimit] = useState<BigNumber | undefined>(undefined);
+  const [newBorrowPercent, setNewBorrowPercent] = useState(
+    userTotalBorrowBalance.div(userTotalBorrowLimit).times(100),
+  );
+  useEffect(() => resetForm, []);
+  useEffect(() => {
+    const tokenPrice = getBigNumber(asset?.tokenPrice);
+    const collateralFactor = getBigNumber(asset?.collateralFactor);
+
+    if (tokenPrice && validAmount) {
+      const amountInUsd = convertWeiToCoins({
+        value: amount as BigNumber,
+        tokenSymbol: asset.id,
+      })
+        .times(tokenPrice)
+        .times(collateralFactor);
+      const temp = newBalance(amountInUsd);
+      setNewBorrowLimit(BigNumber.maximum(temp, 0));
+      setNewBorrowPercent(userTotalBorrowBalance.div(temp).times(100));
+    } else if (BigNumber.isBigNumber(userTotalBorrowLimit)) {
+      setNewBorrowLimit(undefined);
+      setNewBorrowPercent(userTotalBorrowBalance.div(userTotalBorrowLimit).times(100));
+    }
+  }, [amount, asset?.id, userTotalBorrowBalance, userTotalBorrowLimit]);
+
+  // Convert wallet/ supply balance into wei
+  const maxInputWei = React.useMemo(
+    () => convertCoinsToWei({ value: maxInput, tokenSymbol: assetId }),
+    [maxInput.toString()],
+  );
+  return (
+    <>
+      <TokenTextField
+        name="amount"
+        tokenSymbol={assetId as TokenId}
+        value={amount}
+        onChange={amt => setFieldValue('amount', amt, true)}
+        maxWei={maxInputWei}
+        rightMaxButtonLabel={t('supplyWithdraw.max').toUpperCase()}
+        css={styles.input}
+      />
+      <Typography component="span" variant="small1" css={styles.greyLabel}>
+        <Trans i18nKey="supplyWithdraw.walletBalance">
+          <Typography component="span" variant="small1" css={styles.whiteLabel}>
+            {{ amount: asset.walletBalance, symbol: assetId?.toUpperCase() }}
+          </Typography>
+        </Trans>
+      </Typography>
```

Comment:
I chose to use the Typography tag because I think that is preferential to using styling text outside of our theme. Two tags are used to style a single translation with different styles.

These are picked up when I run the command `yarn extract-translation-keys`.

I found this syntax for notating tags in translations which is why I went down this route.
[Source](https://react.i18next.com/latest/trans-component#usage-with-simple-html-elements-like-less-than-br-greater-than-and-others-v10.4.0)